dali/kernels/signal/resampling.h

```diff
@@ -54,19 +59,23 @@ inline float32x4_t vsetq_f32(float x0, float x1, float x2, float x3) {
 #endif
 
 struct ResamplingWindow {
-  inline std::pair<int, int> input_range(float x) const {
+  struct InputRange {
+    int i0, i1;
+  };
+
+  inline DALI_HOST_DEV InputRange input_range(float x) const {
     int xc = std::ceil(x);
     int i0 = xc - lobes;
     int i1 = xc + lobes;
     return {i0, i1};
   }
 
-  inline float operator()(float x) const {
+  inline DALI_HOST_DEV float operator()(float x) const {
```

Comment:
Maybe this deserves a comment or a doc to explain why it is this way. For me it was not clear from the code, why it is done this way. I know you did not change how it works, but since you are touching it :)